code/lighting/lighting_profiles.cpp

```diff
@@ -8,6 +8,134 @@
 
+//*************************************************
+//			lighting_profile_value funcs
+//*************************************************
+
+/**
+ * @brief Processes an input according to the user's configuration and returns the result.
+ */
+float lighting_profile_value::handle(float input)
+{
+	if (only_positive && input < 0.0f)
+		input = base;
+	if (has_adjust)
+		input += adjust;
+	if (has_multiplier)
+		input *= multipier;
+	if (has_minimum)
+		input = MAX(input, minimum);
+	if (has_maximum)
+		input = MIN(input, maximum);
+	return input;
+}
+
+void lighting_profile_value::reset()
+{
+	base = 1.0f;
+	has_adjust = false;
+	has_minimum = false;
+	has_maximum = false;
+	has_multiplier = false;
+	only_positive = true;
+}
+
+void lighting_profile_value::set_adjust(float in)
+{
+	has_adjust = true;
+	adjust = in;
+}
+
+void lighting_profile_value::set_multiplier(float in)
+{
+	has_multiplier = true;
+	multipier = in;
+}
+
+void lighting_profile_value::set_maximum(float in)
+{
+	has_adjust = true;
+	adjust = in;
+}
+
+void lighting_profile_value::set_minimum(float in)
+{
+	has_adjust = true;
+	adjust = in;
+}
+/**
+ * @brief for use during the parsing of a light profile to attempt to read in an LPV
+ *
+ * @param filename for error reporting primairly
+ * @param valuename Text of the field 
+ * @param profile_name The name of the profile being parsed, for error reporting
+ * @param value_target The profile object parsing into
+ * @param required If false it is an error for this to not find a value.
+ * @return true if a succesful parse, false otherwise
+ */
+bool lighting_profile_value::parse(const char* filename,
+	const char* valuename,
+	const SCP_string& profile_name,
+	lighting_profile_value* value_target,
+	bool required)
+{
+	if (optional_string(valuename)) {
+		bool parsed = true;
+		int parses = 0;
+		while (parsed) {
+			parsed = false;
+			if (parse_optional_float_into("+default:", &value_target->base)) {
+				parsed = true;
+				parses++;
+			}
+			if (parse_optional_float_into("+maximum:", &value_target->maximum)) {
+				value_target->has_maximum = true;
+				parsed = true;
+				parses++;
+			}
+			if (parse_optional_float_into("+minimum:", &value_target->minimum)) {
+				value_target->has_minimum = true;
+				parsed = true;
+				parses++;
+			}
+			if (parse_optional_float_into("+multiplier:", &value_target->multipier)) {
+				value_target->has_multiplier = true;
+				parsed = true;
+				parses++;
+			}
+			if (parse_optional_float_into("+adjust:", &value_target->adjust)) {
+				value_target->has_adjust = true;
+				parsed = true;
+				parses++;
+			}
+		}
+		if (parses == 0) {
+			Warning(LOCATION,
+				"Lighting profile value '%s' in file '%s' profile '%s' parsed but set no properties, possible "
+				"malformed table.",
+				valuename,
+				filename,
+				profile_name.c_str());
+		} else if (parses > 5) {
+			Warning(LOCATION,
+				"Lighting profile value '%s' in file '%s' profile '%s' parsed too man no properties, possible "
```

Comment:
"too man no"?

Reply:
I wonder how tired I was when writing that one, woof.